src/core/recv_buffer.c

```diff
@@ -670,7 +843,6 @@ QuicRecvBufferRead(
                 Link);
         CXPLAT_DBG_ASSERT(!Chunk->ExternalReference);
         CXPLAT_DBG_ASSERT(RecvBuffer->ReadStart == 0);
-        CXPLAT_DBG_ASSERT(*BufferCount >= 1);
```

Comment:
Instead of deleting this, should be simply add a `|| RecvBuffer->RecvMode == APP_OWNED`?